cpp/autosar/src/rules/A13-3-1/FunctionThatContainsForwardingReferenceAsItsArgumentOverloaded.ql

```diff
@@ -21,12 +21,26 @@ class Candidate extends TemplateFunction {
   }
 }
 
-from Candidate c, Function f
+from Candidate c, Function f, Function overload, Function overloaded, string msg
 where
   not isExcluded(f,
     OperatorsPackage::functionThatContainsForwardingReferenceAsItsArgumentOverloadedQuery()) and
   not f.isDeleted() and
   f = c.getAnOverload() and
-  //allow for overloading with different number of parameters
-  f.getNumberOfParameters() = c.getNumberOfParameters()
-select f, "Function overloads a $@ with a forwarding reference parameter.", c, "function"
+  // allow for overloading with different number of parameters, because there is no
+  // confusion on what function will be called.
+  f.getNumberOfParameters() = c.getNumberOfParameters() and
+  //build a dynamic select statement that guarantees to read that the overloading function is the explicit one
+  if
+    (f instanceof CopyConstructor or f instanceof MoveConstructor) and
+    f.isCompilerGenerated()
+  then (
+    msg = "implicit constructor" and
+    overloaded = f and
+    overload = c
+  ) else (
+    msg = "function" and
+    overloaded = c and
+    overload = f
+  )
+select overload, "Function overloads a $@ with a forwarding reference parameter.", overloaded, msg
```

Comment:
The alert message requires some further explanation depending on the case.
1. "Function with forward reference parameter overloads implict constructor"
2. "Function overloads a ..."

Reply:
I was also expecting a change in the expected file due to this change in alert location.

Reply:
yep!